rct229/rulesets/ashrae9012019/ruleset_functions/get_swh_components_associated_with_each_swh_bat.py

```diff
@@ -0,0 +1,106 @@
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_energy_required_to_heat_swh_use import (
+    get_energy_required_to_heat_swh_use,
+)
+
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.pint_utils import ZERO
+from rct229.utils.utility_functions import (
+    find_exactly_one_service_water_heating_use,
+    find_exactly_one_service_water_heating_distribution_system,
+)
+
+
+@dataclass
+class SWHEquipmentAssociations:
+    energy_required: ZERO.ENERGY
+    swh_distribution: List[str] = field(default_factory=list)
+    swh_heating_eq: List[str] = field(default_factory=list)
+    pumps: List[str] = field(default_factory=list)
+    tanks: List[str] = field(default_factory=list)
+    piping: List[str] = field(default_factory=list)
+    solar_thermal: List[str] = field(default_factory=list)
+    swh_uses: List[str] = field(default_factory=list)
+
+
+def get_swh_components_associated_with_each_swh_bat(
+    rmd: dict, is_leap_year: bool
+) -> dict[str, SWHEquipmentAssociations]:
+    """
+    This function gets all the SWH equipment connected to a particular SWH use type.  The information is stored in a dictionary where the keys are the ServiceWaterHeatingSpaceOptions2019ASHRAE901 and values are a dictionary giving the ServiceWaterHeatingDistributionSystem, ServiceWaterHeatingEquipment, and Pumps connected to the particular use type.  There is also an entry for the total energy required to heat all SWH uses for a year: "EnergyRequired" .
+
+    Parameters
+    ----------
+    rmd: dict, RMD at RuleSetModelDescription level
+    is_leap_year: bool, default: False
+        Whether the year is a leap year or not.
+    Returns
+    -------
+    swh_and_equip_dict: A dictionary containing where the keys are the ServiceWaterHeatingSpaceOptions2019ASHRAE901 in the RMD and values are dictionaries where keys are the type of SWH equipment and values are the ids of the connected equipment.
+                        Example:  {"DORMITORY":{"swh_distribution":["swhd1","swhd2"], "swh_heating_eq":["swh_eq1","swh_eq2"], "pumps":["p1"], "tanks":["t1"], "piping":["piping1"], "solar_thermal":[], "swh_uses":["swh_use1","swh_use2"], "energy_required": 100000 Btu}}
+    """
+
+    swh_and_equip_dict = {}
+    for building_segment in find_all("$.buildings[*].building_segments[*]", rmd):
+        swh_bat = get_building_segment_swh_bat(rmd, building_segment["id"])
+        swh_and_equip_dict[swh_bat] = SWHEquipmentAssociations(
+            energy_required=ZERO.ENERGY
+        )
+
+        for swh_use_id in find_all(
+            f'$.buildings[*].building_segments[*][?(@.id="{building_segment["id"]}")].zones[*].spaces['
+            f"*].service_water_heating_uses[*].id",
+            rmd,
+        ):
+            swh_use = find_exactly_one_service_water_heating_use(rmd, swh_use_id)
+            energy_required = get_energy_required_to_heat_swh_use(
+                swh_use_id, rmd, building_segment["id"], is_leap_year
+            )
```

Comment:
I think we can loop the `swh_use` object and take out the `id` from the `swh_use` object.